src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java

```diff
@@ -190,64 +182,28 @@ public Action createAdHocAction(final Action action) {
     return actionRepo.saveAndFlush(action);
   }
 
-  @Transactional(propagation = Propagation.REQUIRES_NEW)
   public void createScheduledActions(Integer actionPlanPk) {
-    // Action plan job has to be created before actions
-    ActionPlanJob actionPlanJob = createActionPlanJob(actionPlanPk);
+    try (Stream<ActionCase> cases = actionCaseRepo.findByActionPlanFK(actionPlanPk)) {
+      List<ActionRule> rules = actionRuleRepo.findByActionPlanFK(actionPlanPk);
```

Comment:
do we need to be selecting actionrules again given we've already selected the specific actionrules earlier on in [processActionPlans](https://github.com/ONSdigital/census-rm-action-service/pull/5/files#diff-a8595ccce103d6bddd5212ca7ceee21bR26) method?

also, i think this will apply all actionrules regardless of whether the triggerDateTime has passed?

Reply:
Have a look at the new logic